src/Controller/Component/RequestHandlerComponent.php

```diff
@@ -300,6 +300,7 @@ public function beforeRedirect(Event $event, $url, Response $response)
                 'REQUEST_METHOD' => 'GET'
             ]
         ]));
+        $response->statusCode(200);
         $response->send();
         $response->stop();
     }
```

Comment:
It breaks BC though, does it not? Returning an object now and handling it from the controller? I wonder if we should merge this and apply the proposed refactoring for 3.1 instead. I would be willing to do that.